usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase/ValidateTripTravelSpeed.java

```diff
@@ -0,0 +1,129 @@
+    private final static float METER_PER_SECOND_TO_KMH_CONVERSION_FACTOR = 3.6f;
+
+    /**
+     * @param dataRepo   a repository storing the data of a GTFS dataset
+     * @param resultRepo a repository storing information about the validation process
+     * @param logger     a logger to log information about the validation process
+     */
+    public ValidateTripTravelSpeed(final GtfsDataRepository dataRepo,
+                                   final ValidationResultRepository resultRepo,
+                                   final GeospatialUtils geoUtils,
+                                   final Logger logger) {
+        this.dataRepo = dataRepo;
+        this.resultRepo = resultRepo;
+        this.geoUtils = geoUtils;
+        this.logger = logger;
+    }
+
+    /**
+     * Use case execution method: Checks all records of `trips.txt` refer to an a collection of {@code StopTime} from file
+     * `stop_times.txt` of which -when sorted by `stop_sequence`- the travel speed between each stop is not above 150 km/h
+     * A new {@link FastTravelBetweenStopsNotice} notice is generated each time this condition is false.
+     * This notice is then added to the {@link ValidationResultRepository} provided in the constructor.
+     */
+    public void execute() {
+        logger.info("Validating rule E046 - Fast travel between stops");
+
+        dataRepo.getTripAll().keySet().forEach(tripId -> {
+            var previousStopsData = new Object() {
+                Integer departureTime = null;
+                double latitude;
+                double longitude;
+
+                // used to accumulate distance between stops with same arrival and departure times
+                int accumulatedDistanceMeter = 0;
+                final List<Integer> accumulatedStopSequence = new ArrayList<>();
+            };
+
+            dataRepo.getStopTimeByTripId(tripId).forEach((stopSequence, stopTime) -> {
+
+                // prepare data for current iteration
+                LocationBase currentStop = dataRepo.getStopById(stopTime.getStopId());
+                double currentStopLat = currentStop.getStopLat();
+                double currentStopLon = currentStop.getStopLon();
+
+                Integer currentArrivalTime = stopTime.getArrivalTime();
+                int distanceFromPreviousStopMeter = geoUtils.distanceBetweenMeter(
+                        previousStopsData.latitude,
+                        previousStopsData.longitude,
+                        currentStopLat,
+                        currentStopLon
+                );
+                boolean sameArrivalAndDeparture = false;
+
+                if (previousStopsData.departureTime != null && currentArrivalTime != null) {
+                    sameArrivalAndDeparture = currentArrivalTime.equals(previousStopsData.departureTime);
+
+                    if (!sameArrivalAndDeparture) {
+                        int durationSecond = currentArrivalTime - previousStopsData.departureTime;
```

Comment:
Also, maybe we could add javadoc stating that it is assumed that the n-th stoptime of a trip has `departure_time` < `arrival_time` of the (n+1) stoptime for the same trip. So that it's clear that  cases where `durationSecond < 0` are not handled at this stage.